app/assets/javascripts/oxalis/model/bucket_data_handling/temporal_bucket_manager.js

```diff
@@ -30,7 +30,12 @@ class TemporalBucketManager {
 
   addBucket(bucket: DataBucket): void {
     this.pullBucket(bucket);
-    this.loadedPromises.push(this.makeLoadedPromise(bucket));
+
+    if (bucket.isMissing()) {
```

Comment:
Saving in a volume tracing with a new segmentation, always failed for me locally, because the buckets were already marked as missing, but were then pushed to the pushqueue again by the temporal bucket manager. However, the pushQueue did not request the buckets again (they were already there, marked as missing) and therefore the `loadedPromise` was never triggered.

Reply:
I also wondered why I could only reproduce this locally and investigated. Turns out, the "Render Missing Data Black" setting influences this. If it is turned OFF, the labeled buckets cannot be saved. I couldn't identify the exact cause of this yet.

However, I agree with your feedback regarding the fix and will change the code accordingly :)

Reply:
Ah, that makes sense. If the "render missing data black" setting is true (which is the default), we don't use the "missing" state, but simply create empty buckets in that case (as it has been before). In general, the "missing" state could probably be refactored a bit, so that more components know about this. For example the texture bucket manager does not know about missing buckets and thus copies these to the GPU even though they only contain zeros. I'll create an issue for that.